drivers/sensor/npm1300_charger/npm1300_charger_trigger.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_npm1300_charger
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/mfd/npm1300.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/sensor/npm1300_charger.h>
+
+#include "npm1300_charger.h"
+
+/* nPM1300 base addresses */
+#define MAIN_BASE 0x00U
+
+/* nPM1300 MAIN register offsets */
+#define MAIN_OFFSET_SET      0x00U
+#define MAIN_OFFSET_CLR      0x01U
+#define MAIN_OFFSET_INTENSET 0x02U
+#define MAIN_OFFSET_INTENCLR 0x03U
+
+#define MAIN_SIZE 0x26U
+
+struct event_reg_t {
+	struct sensor_trigger trig;
+	uint8_t offset;
+	uint8_t mask;
+};
+
+static const struct event_reg_t event_reg[NPM1300_TRIGGER_HANDLERS] = {
+	/* Charge completed event */
+	{{SENSOR_TRIG_THRESHOLD, SENSOR_CHAN_NPM1300_CHARGER_STATUS}, 0x0AU, 0x10U},
+	/* Charge error event */
+	{{SENSOR_TRIG_THRESHOLD, SENSOR_CHAN_NPM1300_CHARGER_ERROR}, 0x0AU, 0x20U},
+	/* Battery detected event */
+	{{SENSOR_TRIG_THRESHOLD, SENSOR_CHAN_GAUGE_VOLTAGE}, 0x0EU, 0x01U},
+	/* Battery removed event */
+	{{SENSOR_TRIG_STATIONARY, SENSOR_CHAN_GAUGE_VOLTAGE}, 0x0EU, 0x02U},
+	/* Shiphold button press event */
+	{{SENSOR_TRIG_TAP, SENSOR_CHAN_NPM1300_SHIPHOLD}, 0x12U, 0x01U},
+	/* Watchdog timer warning event */
+	{{SENSOR_TRIG_THRESHOLD, SENSOR_CHAN_NPM1300_TIMER}, 0x12U, 0x08U},
+	/* VBUS detection event */
+	{{SENSOR_TRIG_THRESHOLD, SENSOR_CHAN_GAUGE_DESIRED_CHARGING_CURRENT}, 0x16U, 0x01U},
+	/* VBUS removed event */
+	{{SENSOR_TRIG_STATIONARY, SENSOR_CHAN_GAUGE_DESIRED_CHARGING_CURRENT}, 0x16U, 0x02U},
```

Comment:
I've overloaded the STATIONARY trigger, but this may be better implemented as a separate DETECT/REMOVE or RISING_EDGE/FALLING_EDGE enum.